File: src/main/java/mk/ukim/finki/emt/accommodationrental/model/domain/Host.java
package mk.ukim.finki.emt.accommodationrental.model.domain;

import jakarta.persistence.*;
import lombok.*;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "hosts")
public class Host extends BaseAuditableEntity
{

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private String surname;

    @ManyToOne(optional = false)
    private Country country;

}

File: accommodation-rental-frontend/src/types/host.ts
import type { Country } from "./country";

export interface Host {
    id: number;
    name: string;
    surname?: string;
    fullName?: string;
    country?: Country;
}

export interface CreateHostRequest {
    name: string;
    surname: string;
    countryId: number;
}

export interface UpdateHostRequest {
    name: string;
    surname: string;
    countryId: number;
}